### PR TITLE
Continuous integration

### LINKED ISSUE #7: Proper Continuous Integration
Right now we're using `vagrant plugin install vagrant-vmck` which pulls the `vagrant-vmck` gem from the public repository.

Use this guide to install the vagrant plugin from local code:

    https://3ofcoins.net/2013/06/10/installing-vagrant-plugins-from-local-file/#solution

The above hack should go into the Dockerfile.

### PR DESCRIPTION
**Pull request by gabriel-v**
This was actually pretty simple to do.

closes https://github.com/vmck/vagrant-vmck/issues/7

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+name: test
+
+steps:
+- name: docker build
+  image: docker:dind
+  volumes:
+  - name: dockersock
+    path: /var/run
+  commands:
+  - sleep 5; until docker version; do sleep 5; done
```
Comment (mgax) [inline .drone.yml]: How long does it take docker to start?!

File: Gemfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 group :development do
-  gem 'vagrant', git: 'https://github.com/hashicorp/vagrant.git'
+  gem 'vagrant', tag: 'v2.2.4', git: 'https://github.com/hashicorp/vagrant.git'
```
Comment (mgax) [inline Gemfile:4]: Don't we just want latest?

File: lib/vagrant-vmck/version.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module VagrantPlugins
   module Vmck
-    VERSION = "0.0.3"
+    VERSION = "0.0.4"
```
Comment (mgax) [inline lib/vagrant-vmck/version.rb]: No, let's make this change in a separate commit, after we merge into master.

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  - docker run vagrant-vmck vagrant plugin list
+  - docker run vagrant-vmck vagrant plugin list | grep -q vagrant-vmck
+
+- name: launch example VM
+  image: docker:dind
+  volumes:
+  - name: dockersock
+    path: /var/run
+  commands:
+  - export VMCK_URL=http://$VMCK_IP:$VMCK_PORT
```
Comment (mgax) [inline .drone.yml:31]: Where do `$VMCK_IP` and `$VMCK_PORT` come from? Do we ever use them separately?

File: Dockerfile [hunk 5]
```diff
@@ -1,13 +1,19 @@
-FROM debian:stretch
+FROM debian:testing
```
Comment (mgax) [inline Dockerfile]: Why testing? Stable feels more... stable :)

File: Dockerfile [hunk 6]
```diff
@@ -1,13 +1,19 @@
...
 
 RUN set -e \
- && apt-get update \
- && apt-get install -y procps curl rsync kmod ssh \
+ && apt-get update -yqq \
+ && apt-get install -yqq git procps curl rsync kmod ssh ruby ruby-dev build-essential \
```
Comment (mgax) [inline Dockerfile:5]: So the image is twice as large now. I don't like it, but if there is no simple way to reduce it again, at least make an issue pls.

Comment (mgax) [reply Dockerfile:5]: One option could be to have two "from"s in the build, and copy over the entire `/opt` folder, after installing vagrant and the plugin.

File: Dockerfile [hunk 7]
```diff
@@ -1,13 +1,19 @@
...
+ && rm vagrant_2.2.4_x86_64.deb \
+ && vagrant --version
 
-RUN vagrant plugin install vagrant-vmck
+RUN mkdir /src
+WORKDIR /src
+ADD . .
+
+RUN gem build vagrant-vmck.gemspec
+RUN vagrant plugin install vagrant-vmck-*.gem
```
Comment (mgax) [inline Dockerfile:19]: Ain't that some magic? Beautiful.

**Review by mgax — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 2 above)
Comment (gabriel-v) [reply Gemfile:4]: No, we want the one we're installing in the Dockerfile.

File: lib/vagrant-vmck/version.rb (hunk 3 above)
Comment (gabriel-v) [reply lib/vagrant-vmck/version.rb]: That's right, sorry, forgot it in here.

File: .drone.yml (hunk 4 above)
Comment (gabriel-v) [reply .drone.yml:31]: You can't make drone put `:` in the environment variables that get passed to the jobs, because they are separated by `:`.

File: Dockerfile (hunk 5 above)
Comment (gabriel-v) [reply Dockerfile]: Doesn't work, ruby too old.

File: Dockerfile (hunk 6 above)
Comment (gabriel-v) [reply Dockerfile:5]: The plugins are hidden away in `/root/.vagrant.d/...`. Sure, [here's an issue](https://github.com/vmck/vagrant-vmck/issues/10).

File: .drone.yml (hunk 1 above)
Comment (gabriel-v) [reply .drone.yml]: Between 3 and 10 seconds, but I don't want intermittent errors.

File: .drone.yml (hunk 4 above)
Comment (mgax) [reply .drone.yml:31]: Ah yes, wonderful.

Comment (mgax) [reply .drone.yml:31]: https://discourse.drone.io/t/setting-global-agent-environment-url-variables/4516 - maybe we should make a PR to them :)

File: Dockerfile (hunk 5 above)
Comment (mgax) [reply Dockerfile]: Then let's base the image on Ubuntul LTS.

File: .drone.yml (hunk 1 above)
Comment (mgax) [reply .drone.yml]: Then how about `sleep 1`? We might get some spam in the logs but that's ok.

Comment (mgax) [reply .drone.yml]: bump :)

File: Dockerfile (hunk 5 above)
Comment (mgax) [reply Dockerfile]: bump?

**Review by mgax — DISMISSED** [review]
You missed a couple.
